transcoder/src/keyframes.go

```diff
@@ -231,6 +242,15 @@ func getVideoKeyframes(path string, video_idx uint32, kf *Keyframe) error {
 
 		ret = append(ret, fpts)
 
+		// Notify listeners that enough keyframes for index generation have been parsed
+		if !indexNotificationComplete && fpts >= minParsedKeyframeTime && len(ret) >= minParsedKeyframeCount {
+			kf.add(ret)
+			done += len(ret)
+			ret = ret[:0]
+			indexNotificationComplete = true
+			close(kf.info.indexKeyframesParsed)
+		}
```

Comment:
You can just update the condition below of
```go
if len(ret) == limit
```
to
 ```go
if (
	(!indexNotificationComplete && fpts >= minParsedKeyframeTime && len(ret) >= minParsedKeyframeCount)
	|| len(ret) == limit
) {
```

kf.add() already handles the wait group & notify the code in `stream.go`